### PR TITLE
Added API `rz_debug_switch_to_first_thread()` Instead of rizin commands

### PR DESCRIPTION
**Pull request by HN026**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

Partially Addresses #382 

...

### CODE REVIEW

File: librz/include/rz_core.h [hunk 1]
```diff
@@ -402,6 +402,7 @@ RZ_API void rz_core_notify_error(RZ_NONNULL RzCore *core, RZ_NONNULL const char
  */
 RZ_API void rz_core_gadget_free(RzCoreGadget *g);
 RZ_API void rz_core_gadget_print(RzCore *core);
+RZ_API void rz_core_debug_switch_to_first_thread(RzCore *core);
```
Comment (wargio) [inline librz/include/rz_core.h]: i wonder if this should be part of `rz_debug`.

Comment (HN026) [reply librz/include/rz_core.h]: Yeah makes sense I'm mostly using the debug API's inside this function, will fix.

Comment (wargio) [reply librz/include/rz_core.h]: just pass `RzDebug *debug` as argument and also write the doxygen and add the `RZ_NONNULL` part (check includes).

Comment (HN026) [reply librz/include/rz_core.h]: I'm trying to test it on this piece of multithreading code. But When I try to list the active threads that are attached to this process, it only lists one. 

```C
#include <stdio.h>
#include <pthread.h>
#include <unistd.h>
#include <sys/types.h>

#define NUM_THREADS 5

void *thread_function(void *arg) {
    printf("Thread %ld is running...\n", (long)arg);
    printf("Thread ID is %lu\n", (unsigned long)pthread_self());
    sleep(10);
    printf("Thread %ld finished.\n", (long)arg);
    pthread_exit(NULL);
}

int main() {
    pthread_t threads[NUM_THREADS];
    int rc;
    long t;

    for(t = 0; t < NUM_THREADS; t++) {
        printf("Creating thread %ld\n", t);
        rc = pthread_create(&threads[t], NULL, thread_function, (void *)t);
        if (rc) {
            printf("ERROR; return code from pthread_create() is %d\n", rc);
            return -1;
        }
    }

    for(t = 0; t < NUM_THREADS; t++) {
        pthread_join(threads[t], NULL);
    }

    return 0;
}
```





```sh
$ sudo rizin -d test
Process with PID 147589 started...
WARNING: Neither hash nor gnu_hash exist. Falling back to heuristics for deducing the number of dynamic symbols...
WARNING: Neither hash nor gnu_hash exist. Falling back to heuristics for deducing the number of dynamic symbols...
WARNING: Neither hash nor gnu_hash exist. Falling back to heuristics for deducing the number of dynamic symbols...
 -- Review all the subcommands of aa to see better ways to analyze your targets.
[0x7f330c02bbc0]> dp
 * 147589 ppid:147587 uid:0 s ./test
[0x7f330c02bbc0]> dpt
current ppid   pid    uid status path   
----------------------------------------
true    147587 147589 0   s      ./test
[0x7f330c02bbc0]> 
```

File: librz/core/tui/visual.c [hunk 2]
```diff
@@ -166,11 +166,15 @@ RZ_IPI void rz_core_visual_applyDisMode(RzCore *core, int disMode) {
 	}
 }
 
-RZ_API void rz_core_debug_switch_to_first_thread(RzCore *core) {
-	RzList *threads = rz_debug_pids(core->dbg, core->dbg->pid);
+/**
+ * @brief Switches to the first thread in the given debug context.
+ * @param debug The debug context.
+ */
+RZ_API void rz_debug_switch_to_first_thread(RZ_NONNULL RzDebug *debug) {
```
Comment (wargio) [inline librz/core/tui/visual.c]: this should go in `librz/debug`

File: librz/include/rz_core.h [hunk 3]
```diff
@@ -402,13 +402,14 @@ RZ_API void rz_core_notify_error(RZ_NONNULL RzCore *core, RZ_NONNULL const char
...
 RZ_API void rz_core_gadget_free(RzCoreGadget *g);
 RZ_API void rz_core_gadget_print(RzCore *core);
-RZ_API void rz_core_debug_switch_to_first_thread(RzCore *core);
 
 RZ_API bool rz_core_plugin_init(RzCore *core);
 RZ_API bool rz_core_plugin_add(RzCore *core, RZ_NONNULL RzCorePlugin *plugin);
 RZ_API bool rz_core_plugin_del(RzCore *core, RZ_NONNULL RzCorePlugin *plugin);
 RZ_API bool rz_core_plugin_fini(RzCore *core);
 
+RZ_API void rz_debug_switch_to_first_thread(RZ_NONNULL RzDebug *debug);
```
Comment (wargio) [inline librz/include/rz_core.h]: this should go in `rz_debug.h`

File: librz/debug/debug.c [hunk 4]
```diff
@@ -1756,3 +1756,16 @@ RZ_API void rz_debug_bp_rebase(RzDebug *dbg, ut64 old_base, ut64 new_base) {
 		bp->delta = bp->addr - dbg->bp->baddr;
 	}
 }
+
+/**
+ * @brief Switches to the first thread in the given debug context.
+ * @param debug The debug context.
```
Comment (wargio) [inline librz/debug/debug.c]:
```suggestion
 * \brief Switches to the first thread in the given debug context.
 * \param debug The debug context.
```

File: librz/debug/debug.c [hunk 5]
```diff
@@ -1756,3 +1756,16 @@ RZ_API void rz_debug_bp_rebase(RzDebug *dbg, ut64 old_base, ut64 new_base) {
...
+ */
+RZ_API void rz_debug_switch_to_first_thread(RZ_NONNULL RzDebug *debug) {
```
Comment (wargio) [inline librz/debug/debug.c:1764]:
```
RZ_API void rz_debug_switch_to_first_thread(RZ_NONNULL RzDebug *debug) {
```
you define `debug` as nonnull, but you never do the API check using `rz_return_if_fail`

Comment (HN026) [reply librz/debug/debug.c:1764]: sorry, I missed it.

**Review by wargio — APPROVED** [review]

**Pull request merged**